qcodes/dataset/descriptions.py

```diff
@@ -0,0 +1,107 @@
+import io
+from typing import Dict, Any
+import json
+
+from qcodes.dataset.dependencies import InterDependencies
+
+
+class RunDescriber:
+    """
+    The object that holds the description of each run in the database. This
+    object serialises itself to a string and is found under the run_description
+    column in the runs table
+
+    Extension of this object is planned for the future, for now it holds the
+    parameter interdependencies. Extensions should be objects that can
+    serialize themselves added as attributes to the RunDescriber , such that
+    the RunDescriber can iteratively serialize its attributes when serializing
+    itself.
+    """
+
+    def __init__(self, interdeps: InterDependencies) -> None:
+
+        if not isinstance(interdeps, InterDependencies):
+            raise ValueError('The interdeps arg must be of type: '
+                             f'InterDependencies. Got {type(interdeps)}.')
+
+        self.interdeps = interdeps
+
+    def serialize(self) -> Dict[str, Any]:
+        """
+        Serialize this object into a dictionary
+        """
+        ser = {}
+        ser['Parameters'] = self.interdeps.serialize()
```

Comment:
I know, it's most probably a minor detail. Still, `Parameters` is capitalized, while `paramspecs` is not. I have a feeling that how these "names" (or identifiers) are written should have some sense since we are defining a DSL, right? I don't see any sense in `Parameters` being capitalized, and `paramspecs` not. But I can see myself implementing something that uses DSL and making stupid mistakes partially ( ;) ) because of the lack of convention in the DSL.

Another distraction is that `paramspecs` items are `ParamSpec`s while `Parameters` items are `InterDependencies`. But this is acceptable because DSL terms do not necessarily have to map 1-to-1 to class names in qcodes.

My suggestions would be:
* use only lowercase in serialization
* rename InterDependencies, but i don't have good ideas, hence this is not a suggestion